include/TimingTools.h

```diff
@@ -98,6 +98,141 @@ namespace Mezzanine
+                MicroBenchmarkResults(MicroBenchmarkResults&) = default;
+                ~MicroBenchmarkResults() = default;
+
+                /// @brief Get an Index that corresponds to the percentile of performance.
+                /// @param Percent Where to reach into the timings. With 1.0 the slowest and 0.0 the fastest.
+                /// @return A index for a value from the Original Timings vector.
+                TimingLists::size_type GetIndexFromPercent(PreciseReal Percent) const;
+                /// @brief Get a value from the original timings that corresponds to a percentage into to it.
+                /// @param Percent Where to reach into the timings. With 1.0 the slowest and 0.0 the fastest.
+                /// @return A value from the Original Timings vector.
+                TimeType GetIndexValueFromPercent(PreciseReal Percent) const;
+
+                /// @brief How many times was the timed item executed/
+                CountType Iterations = 0;
+                /// @brief How much was the total runtime with as much of the benchmark removed as possible.
+                TimeType Total = TimeType{0};
+                /// @brief The mean execution time; the Total time divided by the number of iterations.
+                TimeType Average = TimeType{0};
+                /// @brief The fastest (fewest time units) execution time. Defaults to one hour
+                TimeType Fastest = TimeType{0};
+                /// @brief The timing that beats out only 1 percent of the others at being the fastest.
+                /// @sa GetIndexValueFromPercent
+                TimeType FasterThan99Percent = TimeType{0};
+                /// @brief The timing that beats out 10 percent of the others at being the fastest.
+                /// @sa GetIndexValueFromPercent
+                TimeType FasterThan90Percent = TimeType{0};
+                /// @brief The meduan execution time; the execution time in the middle.
+                TimeType Median = TimeType{0};
+                /// @brief The timing that beats out 90 percent of the others at being the fastest.
+                /// @sa GetIndexValueFromPercent
+                TimeType FasterThan10Percent = TimeType{0};
+                /// @brief The timing that beats out 99 percent of the others at being the fastest.
+                /// @sa GetIndexValueFromPercent
+                TimeType FasterThan1Percent = TimeType{0};
+                /// @brief The slowest (most time units) execution time.
+                TimeType Slowest = TimeType{0};
+                /// @brief The raw times gathered by a test, sorted by performance.
+                TimingLists SortedTimings;
+                /// @brief The unsorted timings to help study caching effects.
+                TimingLists UnsortOriginalTimings;
+
+            };
+        RESTORE_WARNING_STATE
+
+        /// @brief Time a single execution of some functor
```

Comment:
Here in America, we end our sentences with periods.

Reply:
But I though "'Mukerica, Fuck Yeah" was not only a complete sentence but also a complete paragraph

Fixed.